js/runtime.js

```diff
@@ -1516,7 +1521,18 @@
             getYear: function(date){
                 return date.getFullYear();
             },
-
+            validateYear: function(inputElement){
+                // we manually check for negative numbers because leaving a
+                // min value of 0 will have the erroneous effect of leaving
+                // a leading 0 that can't be deleted when deleting all
+                // characters from the field
+                if (inputElement.value < 0)
```

Comment:
This is comparing `inputElement.value` (a string) with integer 0, which is probably not what you want.